File: dotnet/Solid/Tests/ISP/Example1/DatabaseStoreTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Solid.ISP.Example1;

namespace Tests.ISP.Example1
{
    [TestClass]
    public class DatabaseStoreTest
    {
        private DatabaseStore _subject;

        [TestInitialize]
        public void Setup()
        {
            _subject = new DatabaseStore();
            _subject.Open();
        }

        [TestCleanup]
        public void Teardown()
        {
            _subject.Close();
        }

        [TestMethod]
        public void Test()
        {
            string result = _subject.ExecuteQuery("advancedQuery");
            Assert.AreEqual("Query result!", result);
        }
    }
}

File: java/isp-start/src/test/java/no/bouvet/solid/isp/DatabaseStoreTest.java
package no.bouvet.solid.isp;

import static org.junit.Assert.assertEquals;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

public class DatabaseStoreTest {

    private DatabaseStore subject;

    @Before
    public void beforeTest() {
        subject = new DatabaseStore();
        subject.open();
    }

    @After
    public void afterTest() {
        subject.close();
    }

    @Test
    public void test() {
        // when
        String result = subject.executeQuery("advancedQuery");

        // then
        assertEquals("Query result!", result);
    }

}
